packages/core/src/editor/BlockNoteEditor.ts

```diff
@@ -698,6 +713,26 @@ export class BlockNoteEditor<
     this._tiptapEditor.dispatch(tr);
   };
 
+  // TO DISCUSS
+  /**
+   * Shorthand to get a typed extension from the editor, by
+   * just passing in the extension class.
+   *
+   * @param ext - The extension class to get
+   * @param key - optional, the key of the extension in the extensions object (defaults to the extension name)
+   * @returns The extension instance
+   */
+  public extension<T extends BlockNoteExtension>(
+    ext: { new (...args: any[]): T } & typeof BlockNoteExtension,
+    key = ext.name()
+  ): T {
+    const extension = this.extensions[key] as T;
+    if (!extension) {
+      throw new Error(`Extension ${key} not found`);
+    }
+    return extension;
+  }
```

Comment:
I'm just gonna skip over any of the extension bit. I think there are better ways to do it, but not pertinent right now to discuss & should be separately discussed & written up